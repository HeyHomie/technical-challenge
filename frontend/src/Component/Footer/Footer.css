.footer {
  background-color: #fff;
  border-top: 1px solid #d0d7de;
  display: flex;
  overflow: hidden;
  padding: 16px 80px;
  justify-content: space-evenly;
}
@media screen and (max-width: 767px) {
  .footer {
    flex-direction: column-reverse;
    margin-top: -20px;
    padding: 16px;
  }
}
.footer-logo {
  align-items: center;
  display: flex;
  justify-content: center;
}
.footer-logo span {
  color: #8b949e;
  font-size: 14px;
  margin-left: 16px;
}
.footer-logo svg {
  fill: #8b949e;
}
.footer-links ul {
  align-items: center;
  display: flex;
  list-style: none;
  justify-content: flex-start;
  padding: 0;
  white-space: nowrap;
}
.footer-links ul a {
  color: var(--blue);
  font-size: 14px;
  font-weight: 300;
  margin-left: 16px;
  text-decoration: none;
}
