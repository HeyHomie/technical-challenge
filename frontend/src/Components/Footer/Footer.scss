.footer {
  background-color: var(--background);
  border-top: 1px solid var(--border);
  display: flex;
  overflow: hidden;
  padding: 16px 80px;

  @media screen and (max-width: 767px) {
    flex-direction: column-reverse;
    margin-top: -20px;
    padding: 16px;
  }

  &-logo {
    align-items: center;
    display:flex;
    justify-content: center;

    span {
      color: var(--primary-muted);
      font-size: 14px;
      margin-left: 16px;
    }
    svg {
      fill: var(--icons);
    }
  }

  &-links {    
    ul {
      align-items: center;
      display: flex;
      list-style: none;
      justify-content: flex-start;
      overflow-x: scroll;
      padding: 0;
      white-space: nowrap;

      a {
        color: var(--blue);
        font-size: 14px;
        font-weight: 300;
        margin-left: 16px;
        text-decoration: none;
      }
    }
  }
}
