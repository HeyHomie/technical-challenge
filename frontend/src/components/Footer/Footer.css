.footer {
  max-width: 1280px;
  margin-right: auto;
  margin-left: auto;
}

.footer .container {
  padding: 40px 0px;
  margin-top: 40px;
  margin-left: 16px;
  margin-right: 16px;
  border-top: 1px solid var(--color-border-primary);
  font-size: 12px;
}

.footer .container ul {
  margin: 0;
  padding: 0;
  list-style-type: none;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin-bottom: 8px;
}

.footer .container ul li {
  margin-right: 16px;
}

.footer .container ul li a {
  color: var(--color-accent-fg);
}

.footer .container .github-logo {
  display: none;
}

@media (min-width: 1012px) {
  .footer .container {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  .footer .container .github-logo {
    display: block;
    color: var(--color-footer-invertocat-octicon);
  }

  .footer .container ul {
    width: 41.66666665%;
    justify-content: space-between;
    margin-bottom: 0;
  }
}
