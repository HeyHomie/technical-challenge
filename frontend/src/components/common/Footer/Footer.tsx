import { MarkGithubIcon } from '@primer/octicons-react'

const Footer = () => (
  <footer className="bottom-0 flex h-[7.25rem] items-center text-center justify-start px-4 text-xs border-t border-accent-br mt-10">
    <span className="flex items-center w-1/6 ">
      <MarkGithubIcon size={24} className="mr-2 text-fg-subtle" />
      <span className="text-secondary">Â© 2021 GitHub, Inc.</span>
    </span>
    <ul className="flex flex-wrap justify-between w-2/3 text-accent-fg">
      <li>
        <a href="https://docs.github.com/en/github/site-policy/github-terms-of-service">
          Terms
        </a>
      </li>
      <li>
        <a href="https://docs.github.com/en/github/site-policy/github-privacy-statement">
          Privacy
        </a>
      </li>
      <li>
        <a href="https://github.com/security">Security</a>
      </li>
      <li>
        <a href="https://www.githubstatus.com/">Status</a>
      </li>
      <li>
        <a href="https://docs.github.com">Docs</a>
      </li>
      <li>
        <a href="https://support.github.com/?tags=dotcom-footer">
          Contact GitHub
        </a>
      </li>
      <li>
        <a href="https://github.com/pricing">Pricing</a>
      </li>
      <li>
        <a href="https://docs.github.com">API</a>
      </li>
      <li>
        <a href="https://services.github.com/">Training</a>
      </li>
      <li>
        <a href="https://github.blog/">Blog</a>
      </li>
      <li>
        <a href="https://github.com/about">About</a>
      </li>
    </ul>
  </footer>
)

export default Footer
