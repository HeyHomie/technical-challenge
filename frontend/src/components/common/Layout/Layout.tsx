import { FC } from 'react'
import { Header } from 'components/common'

const Layout: FC = ({ children }) => {
  return (
    <div className="dark">
      <div className="w-screen min-h-screen text-primary bg-primary">
        <Header />
        {children}
      </div>
    </div>
  )
}

export default Layout
