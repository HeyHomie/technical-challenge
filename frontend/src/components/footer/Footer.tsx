import { FC } from 'react'

export const Footer: FC = () => {
  return (
    <footer>
      <p>The footer is here</p>
    </footer>
  )
}
