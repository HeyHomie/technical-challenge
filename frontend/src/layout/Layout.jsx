import React from 'react'
import { useGetData } from '../hooks/useGetData'

export const Layout = ({ children }) => {
  return (
    <>
      <header></header>
      <main>{children}</main>
      <footer></footer>
    </>
  )
}
