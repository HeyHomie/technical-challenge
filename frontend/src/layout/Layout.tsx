import React, { FC } from 'react'
import { Header } from '../components/header/Header'
import { Footer } from '../components/footer/Footer'
import { Main } from '../components/main/Main'

export const Layout: FC = ({ children }) => {
  return (
    <>
      <Header />
      <Main>{children}</Main>
      <Footer />
    </>
  )
}
