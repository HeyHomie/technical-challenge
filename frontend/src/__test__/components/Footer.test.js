import React from 'react'
import { render  } from '@testing-library/react'
import { Footer } from '../../components/Layout/Footer'
import { create } from 'react-test-renderer'

describe('Footer', () => {
  const footer = render(<Footer />)
  test('Render of the component Footer', () => {
    expect(footer.length).toEqual(1);
  })
});