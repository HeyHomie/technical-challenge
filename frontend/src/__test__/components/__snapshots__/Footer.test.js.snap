// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot Comprobate UI from the component Footer 1`] = `
<div
  className="Footer--container"
>
  <div
    className="Footer__info"
  >
    <p>
      Terms
    </p>
    <p>
      Privacy
    </p>
    <p>
      Security
    </p>
    <p>
      Status
    </p>
    <p>
      Docs
    </p>
    <p>
      Contact
    </p>
    <p>
      Pricing
    </p>
    <p>
      API
    </p>
    <p>
      Training
    </p>
    <p>
      Blog
    </p>
    <p>
      About
    </p>
  </div>
  <div
    className="Footer__logo"
  >
    <img
      alt="GitHub Logo"
      src="https://cdn-icons-png.flaticon.com/512/25/25231.png"
    />
    <span>
      Â© 2021 GitHub, Inc.
    </span>
  </div>
</div>
`;
